src/main/java/de/retest/recheck/configuration/ProjectConfiguration.java

```diff
@@ -99,4 +105,7 @@ private InputStream getInputStreamFrom( final String fileName ) {
 		return getClass().getClassLoader().getResourceAsStream( fileName );
 	}
 
+	public void resetConfiguration() {
```

Comment:
Instead of introducing this `public` method, we could do something like this in tests:

```java
@Before
public void setUp() throws Exception {
    Field projectConfigFolder = ProjectConfiguration.class.getDeclaredField("projectConfigFolder");
    projectConfigFolder.setAccessible( true );
    projectConfigFolder.set( null, null );
}
```

Reply:
True. I just assumed that the public method was less of a hack and maybe someone else needs it at some point. Resetting this is not expensive, it just produces another line in the log...

Reply:
>[…] maybe someone else needs it at some point.

But for now it's just test code, isn't it?

This also introduces an `Optional` field (which is controversial in general) that is `null`, this somewhat defeats the purpose of `Optional`.

If you want to keep the reset method, maybe have a nullable `Path` field?

Reply:
For now it's just test code, but a reset is still better than Reflection.
The `null` Optional is genius, because it displays the reason Optional is better than pure `null`. `Optional.empty` means we looked it up and don't know, whereas null means we didn't look it up yet.
But I will better document that...